sling.go

```diff
@@ -145,6 +146,20 @@ func (s *Sling) Set(key, value string) *Sling {
 	return s
 }
 
+// SetBasicAuth mimics http.Request.SetBasicAuth()
+// setting the basic auth headers for all requests
+// which is convenient when building API clients
+func (s *Sling) SetBasicAuth(username, password string) *Sling {
+	return s.Set("Authorization", "Basic "+basicAuth(username, password))
+}
+
+// basicAuth copied from src/net/http/client.go
```

Comment:
// basicAuth returns the base64 encoded username:password for basic auth copied
// from net/http.